api-specs/api_spec_PGS.yaml

```diff
@@ -395,65 +408,150 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/XPayPaymentAuthorizationError'
+  /payment-authorizations/xpay/{paymentAuthorizationId}:     
+    get:       
+      summary: Retrieve xpay payment authorization
+      description:  Retrieve xpay payment authorization given paymentAuthorizationId
+      tags:         
+        - XPay
+      operationId: get-auth-payment-xpay
+      parameters:         
+      - in: path           
         required: true
+        name: paymentAuthorizationId
+        description: Id of the payment authorization
+        example: "41bc2409-5926-4aa9-afcc-797c7054e467"           
+        schema:
+          type: string
+      - in: header
+        name: MDC-Fields
+        description: MDC information
+        example: 97g10t83x7bb0437bbc50sdf58e970gt
+        schema:
+          type: string
+        required: false
       responses:
         "200":
           description: OK
-          content:
-            '*/*':
+          content:             
+            application/json:
               schema:
-                $ref: '#/components/schemas/XPayAuthResponseEntity'
-        "400":
-          description: Bad request - missing mandatory parameters
+                $ref: '#/components/schemas/XPayPaymentAuthorization'     
+        "404":           
+          description: payment authorization not found           
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/XPayAuthResponseEntity400'
-        "401":
-          description: transactionId already processed
+                $ref: '#/components/schemas/XPayPaymentAuthorizationError'
+        "500":
+          description: Internal server error
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/XPayAuthResponseEntity401'
-  /request-payments/xpay/{requestId}/resume:
-    get:
-      summary: payment request to xPay
+                $ref: '#/components/schemas/XPayPaymentAuthorizationError' 
+    delete:
       tags:
         - XPay
-      operationId: resumeXPayPayment
+      summary: refund xpay requests
+      description:  Refund xpay payment authorization given paymentAuthorizationId
+      operationId: refund-auth-payment-xpay
       parameters:
-        - name: requestId
-          in: path
-          required: true
+        - in: path
+          name: paymentAuthorizationId
           schema:
             type: string
-        - name: esito
+            format: uuid
+          required: true
+          description: PGS-generated GUID of the request to retrieve
+          example: 77e1c83b-7bb0-437b-bc50-a7a58e5660ac
+      responses:
+        "200":
+          description: OK
+          content:             
+            application/json:
+              schema:
+                $ref: '#/components/schemas/XPayPaymentAuthorization'     
```

Comment:
Is here the `XPayPaymentAuthorization` wanted as response body? It contains many optional fields that will not be used in the response, such as html. Maybe here we can put a stricter response with only the required fields as refund response?

Reply:
correct!